### ISSUE TITLE
Issue with non 'taggit' field called 'tags'

### ISSUE THREAD

**Issue by hammygoonan** [wagtail/wagtail#3665]
### Issue Summary

I'm using the `CheckboxSelectMultiple` widget for a `ParentalManyToManyField` called `tags`. The `tags` attribute does not use the taggit module

When I go to add an image, the taggit module appears to change the `tags` field on the page to a `'taggit'` field.

### Steps to Reproduce

1. Create a Page model that has an Image field:

```
image = models.ForeignKey(
        'wagtailimages.Image',
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name='+'
    )
```

and a tags field as a foreign key:

```tags = ParentalManyToManyField('tag.Tag', blank=True)```

and their respective panels:

```
content_panels = Page.content_panels + [
    ImageChooserPanel('featured_image'),
    FieldPanel('tags', widget=forms.CheckboxSelectMultiple)
]
```

2. Add some tags to the tag field

3. go into the admin. The `tags` field will be displaying correctly as checkboxes

4. Click 'Choose and image' on the image input.

The `tags` field will now look like a `taggit` field.

Looking through the markup, both the tag field for the image and the tag field for the page have an id of `id_tags`.

My suspicion is that it's this file here causing the issue:

[https://github.com/wagtail/wagtail/blob/c6666c6de5e83bf94d18324858c121e6584ba47d/wagtail/wagtailimages/templates/wagtailimages/chooser/chooser.js#L107](https://github.com/wagtail/wagtail/blob/c6666c6de5e83bf94d18324858c121e6584ba47d/wagtail/wagtailimages/templates/wagtailimages/chooser/chooser.js#L107)

But there are quite a few mentions of $('#id_tags') in the code base:

[https://github.com/wagtail/wagtail/search?utf8=%E2%9C%93&q=id_tags&type=](https://github.com/wagtail/wagtail/search?utf8=%E2%9C%93&q=id_tags&type=)

### Technical details

* Python version: Python 3.5.2
* Django version: Version: 1.11.1
* Wagtail version: Version: 1.10.1
* Browser version: Chrome 58

**Comment by Temoeri** [comment]
Hi @hammygoonan !

I encountered the same problem as you did.
As mentioned in https://docs.djangoproject.com/en/2.0/ref/forms/widgets/#django.forms.Widget, you can give your fields attributes.

I have added a custom ID to my forms.CheckboxSelectMultiple, for example:
`FieldPanel('tags', widget=forms.CheckboxSelectMultiple(attrs={'id': 'some_new_id'})),`

This seems to solve the issue.

Best Regards,

Temoeri

**Comment by coredumperror** [comment]
I ran into the same problem, and used the same solution to fix it. 

However, this should absolutely not be necessary. ImageChooserPanel should be responsible for not screwing up the user's form. Perhaps it could change its field's HTML id to something less generic than "id_tags", like "id_image_chooser_tags".

The problem arrises from this line in the ImageChooser widget:
```
<script>initTagField('id_tags', '/admin/tag\u002Dautocomplete/', true);</script>
```
It selects both the `<input type="text" name="tags" id="id_tags" style="display: none;">` element from the widget, and any other widget on the form called `id_tags`. Another possible solution would be to make that selector more specific. Perhaps `.modal #id_tags`?

**Comment by gasman** [comment]
I believe this is fixed by #5298 (which will be part of Wagtail 2.6).

**Comment by coredumperror** [comment]
That's great!
